[SW-259] Refactor: replace RxJS with Vue Composables

peronczyk opened the pull request.

CedrikNikita commented:
For the new account we are getting such errors, which suppose to be handled as a valid behaviour.
`Uncaught (in promise) Error: v3/accounts/ak_Kqz3NbLUHsmxSB7tAqh4ixq44Hq6rAfSjXNCLP226zABFErdh error: Account not found
    at http (index.js?4f8d:103:1)
    at async request (swagger.mjs?0a14:148:1)
    at async Object.balance (node.mjs?63e1:138:1)
    at async Promise.all (index 0)
`

CedrikNikita reviewed (changes requested):
For the new account we are getting such errors, which suppose to be handled as a valid behaviour.
`Uncaught (in promise) Error: v3/accounts/ak_Kqz3NbLUHsmxSB7tAqh4ixq44Hq6rAfSjXNCLP226zABFErdh error: Account not found at http (index.js?4f8d:103:1) at async request (swagger.mjs?0a14:148:1) at async Object.balance (node.mjs?63e1:138:1) at async Promise.all (index 0)
`

CedrikNikita commented on src/popup/components/TotalWalletAmount.vue:
```diff
@@ -10,27 +10,27 @@
 
     return {
+      balances,
+      balaX,
```
Is`balaX` variable actually used?

peronczyk replied:
My mistake. I was using it for debugging purposes. I'll remove it.

peronczyk commented:
> For the new account we are getting such errors, which suppose to be handled as a valid behaviour. `Uncaught (in promise) Error: v3/accounts/ak_Kqz3NbLUHsmxSB7tAqh4ixq44Hq6rAfSjXNCLP226zABFErdh error: Account not found at http (index.js?4f8d:103:1) at async request (swagger.mjs?0a14:148:1) at async Object.balance (node.mjs?63e1:138:1) at async Promise.all (index 0) `

True, I fixed this. Please see now @CedrikNikita .

CedrikNikita reviewed (changes requested):
Changes made in `package-lock.json` also need to be committed.

peronczyk commented:
> Changes made in `package-lock.json` also need to be committed.

Done

peronczyk commented on src/composables/balances.ts:
```diff
@@ -0,0 +1,93 @@
+type Balances = Record<string, BigNumber>;
+type BalancesRaw = Record<string, string>;
+
+interface IUseBalancesOptions {
```
Yeah, you have right. I'll change this.

davidyuk commented on src/composables/topHeader.ts:
```diff
@@ -0,0 +1,35 @@
+
+interface UseTopHeaderOptions {
+  /**
+   * TODO: Temporary solution to avoid dependency circle
```
Recently I had a dependency cycle in types, and I think it is ok to import such dependency using `import type` then linter won't blame it. This would be a cycle dependency only on the stage of type checking. Would it be ok or would it complicate unit testing or so?

davidyuk replied:
It is possible to import not the whole store but only it's type using `import type`?

davidyuk commented on src/popup/components/TransferSendForm.vue:
```diff
@@ -250,7 +252,7 @@ export default defineComponent({
     );
     const isMaxValue = computed<boolean>(
-      () => isNumbersEqual(formModel.value.amount, max.value),
+      () => isNumbersEqual(+formModel.value.amount, +max.value),
```
```suggestion
      () => +formModel.value.amount === +max.value,
```
🤔

peronczyk replied:
Makes sense as those 2 are not BigNumbers. Changed...

CedrikNikita reviewed (commented):
Why do you replace `useGetter` in some places with `computed` and in other places you do opposite?
It seems that you decided to replace them in `TransactionDetails`, because it is harder to unit test such things.

CedrikNikita reviewed (approved):
In general, code changes look resonable.

peronczyk commented:
> Why do you replace `useGetter` in some places with `computed` and in other places you do opposite? It seems that you decided to replace them in `TransactionDetails`, because it is harder to unit test such things.

Yes, in some places I removed it because it was causing dependency cycle. The file that defines `useGetter` and `useState` imports the store. So when I use them in a composable and then import this composable in any non Vue file ESLint is telling me that there is a dependency cycle.

Also for the `TransactionDetails` i replaced the `useGetter` with `computed` to make the unit test file much simpler and avoid mocking functions.
